src/main/java/seedu/rc4hdb/model/resident/predicates/AttributesMatchKeywordsPredicate.java

```diff
@@ -10,13 +11,53 @@
  */
 public class AttributesMatchKeywordsPredicate implements Predicate<Resident> {
     private final ResidentDescriptor descriptor;
+    private final FilterSpecifier specifier;
 
-    public AttributesMatchKeywordsPredicate(ResidentDescriptor keywords) {
+    /**
+     * Constructor to create predicate object for filter
+     * @param keywords the description to filter the residents with
+     * @param specifier the type of filtering specified
+     */
+    public AttributesMatchKeywordsPredicate(ResidentDescriptor keywords, FilterSpecifier specifier) {
         this.descriptor = keywords;
+        this.specifier = specifier;
     }
 
     @Override
     public boolean test(Resident resident) {
+        switch(specifier.getSpecifier()) {
+        case "all":
+            return filterAll(resident);
+        case "any":
+            return filterAny(resident);
+        default:
+            return false;
+        }
+
+    }
+
+    /**
+     * Filters residents if any description matches
+     * @param resident the resident to verify if they match the filter description
+     * @return if the resident matches the description requirements
+     */
+    public boolean filterAny(Resident resident) {
+        return descriptor.getName().map(name -> name.equals(resident.getName())).orElse(false)
+                || descriptor.getPhone().map(phone -> phone.equals(resident.getPhone())).orElse(false)
+                || descriptor.getEmail().map(email -> email.equals(resident.getEmail())).orElse(false)
+                || descriptor.getRoom().map(room -> room.equals(resident.getRoom())).orElse(false)
+                || descriptor.getGender().map(gender -> gender.equals(resident.getGender())).orElse(false)
+                || descriptor.getHouse().map(house -> house.equals(resident.getHouse())).orElse(false)
+                || descriptor.getMatricNumber().map(matric -> matric.equals(resident.getMatricNumber())).orElse(false)
+                || descriptor.getTags().map(tags -> tags.equals(resident.getTags())).orElse(false);
+    }
```

Comment:
Actually, I think that our filter command might be too strict and this might hinder usability. Perhaps instead of checking for 100% match in each field, i.e. `filter /any n/Jo` will not find John, and will **ONLY** find Jo. Perhaps we could be more lenient with the filter, by checking if Jo is contained in John and other entries.